### PR TITLE
ENH: Add CI configuration files.

### PR DESCRIPTION
**Pull request by jhlegarreta**
Add CircleCI configuration files and badges.

### CODE REVIEW

**Comment by jhlegarreta** [comment]
@richardbeare @hjmjohnson  @fbudin69500  @thewtex  @zachary-williamson Please, review and merge if appropriate.

Specifically, I ignore whether `scikit` is necessary in `setup.py`, and the `version` and `Development Status` fields were just pulled from another remote. Also, @richardbeare I've put your Monash University mail address.

Thanks.

**Comment by richardbeare** [comment]
Hi,
Thanks for this. I'll merge with my repo soon. However, Monash has changed address formats - it should be @monash.edu

**Comment by jhlegarreta** [comment]
Thanks @thewtex ! 5741a60 adds the `CTestConfig.cmake` file. I'll address this for the rest of the remotes that Ive been fiddling with concerning the CI.

**Comment by jhlegarreta** [comment]
`travis-ci` reports an error:
```
CMake Error at wrapping/itkMorphologicalDistanceTransformImageFilter.wrap:1 (WRAP_CLASS):
  Unknown CMake command "WRAP_CLASS".
Call Stack (most recent call first):
  /Users/kitware/Dashboards/ITK/ITKPythonPackage/standalone-build/ITK-source/Wrapping/TypedefMacros.cmake:214 (include)
  /Users/kitware/Dashboards/ITK/ITKPythonPackage/standalone-build/ITK-source/Wrapping/TypedefMacros.cmake:195 (itk_load_submodule)
  wrapping/CMakeLists.txt:2 (itk_auto_load_submodules)
```

Is this related to some bad command in the `.wrap` file or else is it related to some missing tweak in the module configuration?

**Comment by jhlegarreta** [comment]
Looks like the Python wrapping has not been correctly made.:
```
CMake Error at wrapping/itkMorphologicalDistanceTransformImageFilter.wrap:1 (WRAP_CLASS):
  Unknown CMake command "WRAP_CLASS".
```

Any suggestion to fix the `.wrap` files?

**Comment by thewtex** [comment]
> Any suggestion to fix the .wrap files?

The wrapping needs to be updated, e.g. `WRAP_CLASS` is `itk_wrap_class`, etc. More information can be found in the Software Guide:

  https://itk.org/ITKSoftwareGuide/html/Book1/ITKSoftwareGuide-Book1ch9.html#x55-1560009.5

**Comment by jhlegarreta** [comment]
Sorry @thewtex for having taken so long time. Let's see if commit 844f0f3 properly addresses the issue.

File: wrapping/itkParabolicDilateImageFilter.wrap [hunk 1]
```diff
@@ -1,3 +1,3 @@
 WRAP_CLASS("itk::ParabolicDilateImageFilter" POINTER_WITH_SUPERCLASS)
```
Comment (thewtex) [inline wrapping/itkParabolicDilateImageFilter.wrap]: `WRAP_CLASS` -> `itk_wrap_class`

File: wrapping/itkParabolicErodeImageFilter.wrap [hunk 2]
```diff
@@ -1,3 +1,3 @@
 WRAP_CLASS("itk::ParabolicErodeImageFilter" POINTER_WITH_SUPERCLASS)
```
Comment (thewtex) [inline wrapping/itkParabolicErodeImageFilter.wrap]: `WRAP_CLASS` -> `itk_wrap_class`

**Comment by jhlegarreta** [comment]
Sorry for the oversights and thanks for the review @thewtex. Let's see if we get them right with c446d63

**Comment by jhlegarreta** [comment]
Threading mechanism issues uncovered by the CI tools hopefully fixed in PR #17.

**Comment by jhlegarreta** [comment]
ddee914 rebased on master.

**Pull request merged**